addon/mixins/in-viewport.js

```diff
@@ -64,164 +65,164 @@ export default Mixin.create({
-    this._bindScrollDirectionListener(window, get(this, 'viewportScrollSensitivity'));
-
-    if (!get(this, 'viewportUseRAF')) {
-      get(this, 'viewportListeners').forEach((listener) => {
-        const { context, event } = listener;
-        this._bindListeners(context, event);
-      });
-    }
-  },
-
-  _addObserverIfNotSpying() {
-    if (!get(this, 'viewportSpy')) {
-      this.addObserver('viewportEntered', this, this._unbindIfEntered);
-    }
+    this._setInitialViewport();
+    // this._addObserverIfNotSpying();
+    // this._bindScrollDirectionListener(window, get(this, 'viewportScrollSensitivity'));
+
+    // if (!get(this, 'viewportUseRAF')) {
+    //   get(this, 'viewportListeners').forEach((listener) => {
+    //     const { context, event } = listener;
+    //     this._bindListeners(context, event);
+    //   });
+    // }
   },
 
-  _setViewportEntered(context = null) {
-    assert('You must pass a valid context to _setViewportEntered', context);
+  // _addObserverIfNotSpying() {
+  //   if (!get(this, 'viewportSpy')) {
+  //     this.addObserver('viewportEntered', this, this._unbindIfEntered);
+  //   }
+  // },
 
-    const element = get(this, 'element');
+  _setViewportEntered(/* context = null */) {
+    // assert('You must pass a valid context to _setViewportEntered', context);
 
-    if (!element) {
+    if (!this.element) {
       return;
     }
 
-    const $contextEl = $(context);
-    const boundingClientRect = element.getBoundingClientRect();
-
-    this._triggerDidAccessViewport(
-      isInViewport(
-        boundingClientRect,
-        $contextEl.innerHeight(),
-        $contextEl.innerWidth(),
-        get(this, 'viewportTolerance')
-      )
-    );
-
-    if (boundingClientRect && get(this, 'viewportUseRAF')) {
-      rAFIDS[get(this, 'elementId')] = window.requestAnimationFrame(
-        bind(this, this._setViewportEntered, context)
-      );
+    const { top, left, bottom, right } = this.viewportTolerance;
+    const options = {
+      root: null, 
+      rootMargin: `${top}px ${right}px ${bottom}px ${left}px`,
+      threshold: this.viewportScrollSensitivity
     }
-  },
-
-  _triggerDidScrollDirection($contextEl = null, sensitivity = 1) {
-    assert('You must pass a valid context element to _triggerDidScrollDirection', $contextEl);
-    assert('sensitivity cannot be 0', sensitivity);
 
-    const elementId = get(this, 'elementId');
-    const lastDirectionForEl = lastDirection[elementId];
-    const lastPositionForEl = lastPosition[elementId];
-    const newPosition = {
-      top: $contextEl.scrollTop(),
-      left: $contextEl.scrollLeft()
-    };
+    this.observer = new IntersectionObserver(bind(this, this._onIntersection), options)
+    this.observer.observe(this.element);
+  },
 
-    const scrollDirection = checkScrollDirection(lastPositionForEl, newPosition, sensitivity);
-    const directionChanged = scrollDirection !== lastDirectionForEl;
+  _onIntersection(entries) {
```

Comment:
This is the IntersectionObserver callback.  We can detect both entering and exiting.

Reply:
`entries` is an Array, but I think we can assume it is one element...